src/Agent.Worker/DiagnosticLogManager.cs

```diff
@@ -105,6 +105,21 @@ public sealed class DiagnosticLogManager : AgentService, IDiagnosticLogManager
                 File.Copy(agentLogFile, destination);
             }
 
+            // Read and add to logs waagent.conf settings on Linux
+            if (PlatformUtil.RunningOnLinux)
+            {
+                executionContext.Debug("Creating waagent file.");
```

Comment:
Please add fallback - so if any error happens during new logic, it won't crush agent, but will print error message in logs.